client/src/app/utils/hover.js

```diff
@@ -0,0 +1,68 @@
+import d3 from './d3-modules';
+//TODO:Subclass this class to work with any kind of node, edge.
+class Hover {
+
+  //Mouse event when you stop hovering over a node in the visualization.
+  //Shows the representative description of the representative on the edge.
+  handleMouseOverNode(d,i) {  // Add interactivity
+
+    //If statement for checking the status of a node.
+    //is it a superpac or a representative.
+    if(d.type == "representative"){
+      if(d.district == ""){
+        d3.select(this.parentNode).selectAll("g > text#"+d.id)
+          .html(()=>{return d.name+" ("+d.party+") senator from "+d.state;});
+      }else{
+        d3.select(this.parentNode).selectAll("g > text#"+d.id)
+          .html(()=>{return d.name+" ("+d.party+") congressman from "+d.state+"'s district "+d.district;});
+      }
+
+
+      d3.select(this.parentNode).selectAll("g > rect#"+d.id)
+        .attr("width",250);
+        //TODO:Could use a bounding box, but I don't know how to do it.
+        //TODO:Make this less coupled with the json data. Essentially remove need for if statements
+    }
+  }
+
+  //Mouse event when you stop hovering over a node in the visualization.
+  //Shows the representative description of the representative on the edge.
+  handleMouseOutNode(d,i) {
+
+    if(d.type == "representative"){
+      d3.select(this.parentNode).selectAll("g > rect#"+d.id)
+        .attr("width",d.bbox.width+10);
```

Comment:
please use display = "none" instead of setting width to 0 as that can cause undesired side effects